packages/fxa-auth-server/lib/routes/totp.js

```diff
@@ -151,6 +162,31 @@ module.exports = (log, db, mailer, customs, config, glean, profileClient) => {
           throw errors.unverifiedSession();
         }
 
+        // Will also use the recovery phone to validate the code.
+        const code = request.payload.code;
+        if (code) {
+          // Client has posted a code, but we need make it was a code that was actually
+          // sent to the client
+          const success = await recoveryPhoneService.confirmCode(uid, code);
+          if (!success) {
+            throw errors.invalidOrExpiredOtpCode();
```

Comment:
I see that this is the same error used below.  They are different types of codes correct?  Might be better for error reporting / debugging if we use different errors.

Reply:
I was on the fence about introducing a new error type. The fact the error was 'invalid or expired' seemed like it still fit to me. I can see the argument for a separate error type though. I'll circle back on this.